packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,25 +5,26 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {Fragment} from 'react';
-
+import React, {
+  useState,
+  useEffect,
+  Fragment,
+  useContext,
+  useRef,
+  useCallback,
+} from 'react';
 import DocusaurusContext from '@docusaurus/context';
 
 import './styles.css';
 
-class Search extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      enabled: true,
-      isExpanded: false,
-    };
-    this.searchBarRef = React.createRef();
-    this.toggleSearchIconClick = this.toggleSearchIconClick.bind(this);
-  }
+const Search = props => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [enabled, setEnabled] = useState(true);
```

Comment:
Can we rename to `isEnabled` and `setIsEnabled` since we're modifying this part?